File: apps/mo1-ios/ios/Sources/Mo1Core/Domain/Capabilities.swift
// Capabilities — UI show/hide from the server-returned capabilities array
// (design §1, §6: "UI は capabilities に従い出し分けのみ"), mirrors
// capabilities.ts. The client NEVER decides authorization; it only mirrors what
// the BFF grants (default deny).
import Foundation

/// Does the resource's capability set include `key`? (default deny.)
public func can(_ capabilities: [PermissionKey], _ key: PermissionKey) -> Bool {
    capabilities.contains(key)
}

/// S4 action editing / S3 event editing gate.
public func canEditEvent(_ capabilities: [PermissionKey]) -> Bool {
    can(capabilities, .eventWrite)
}

/// S5 task status change gate.
public func canWriteTask(_ capabilities: [PermissionKey]) -> Bool {
    can(capabilities, .taskWrite)
}

File: apps/mo1-ios/src/capabilities.ts
// capabilities — UI show/hide from the server-returned capabilities array
// (design §1, §6: "UI は capabilities に従い出し分けのみ", myPermissions は廃止).
// The client NEVER decides authorization; it only mirrors what the BFF grants.
import type { identity } from "@dub/types";

export type Capability = identity.PermissionKey;

/** Does the resource's capability set include `key`? (default deny.) */
export function can(capabilities: readonly Capability[], key: Capability): boolean {
  return capabilities.includes(key);
}

/** S4 action editing / S3 event editing gate. */
export function canEditEvent(capabilities: readonly Capability[]): boolean {
  return can(capabilities, "event:write");
}

/** S5 task status change gate. */
export function canWriteTask(capabilities: readonly Capability[]): boolean {
  return can(capabilities, "task:write");
}
